src/promptflow/promptflow/_utils/tool_utils.py

```diff
@@ -131,13 +131,22 @@ def function_to_interface(
         }
     )
     # Resolve inputs to definitions.
+    input_index = 0
     for k, v in all_inputs.items():
         # Get value type from annotation
         value_type = resolve_annotation(v.annotation)
         if skip_prompt_template and value_type is PromptTemplate:
             # custom llm tool has prompt template as input, skip it
             continue
+
         input_def, is_connection = param_to_definition(v, gen_custom_type_conn=gen_custom_type_conn)
+        # Set ui_hints and index
+        if add_ui_hints:
```

Comment:
Since we have already introduce this concept, I think it would better always gen this property instead of adding this parameter.

Reply:
Only add this for custom_llm tool because only the custom_llm tool has such a requirement for now. To avoid unexpected changes, other types of tools will remain as they are. Thanks.